tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java

```diff
@@ -301,4 +308,91 @@ public void testLog4j2ModuleWithSimpleWebAppWithJSP() throws Exception
             IO.delete(jettyBase.toFile());
         }
     }
+
+    @Test
+    public void testWebsocketClientInWebapp() throws Exception
+    {
+        Path jettyBase = Files.createTempDirectory("jetty_base");
+        String jettyVersion = System.getProperty("jettyVersion");
+        DistributionTester distribution = DistributionTester.Builder.newInstance()
+            .jettyVersion(jettyVersion)
+            .jettyBase(jettyBase)
+            .mavenLocalRepository(System.getProperty("mavenRepoPath"))
+            .build();
+
+        String[] args1 = {
+            "--create-startd",
+            "--approve-all-licenses",
+            "--add-to-start=resources,server,http,webapp,deploy,jsp,jmx,servlet,servlets,websocket"
+        };
+        try (DistributionTester.Run run1 = distribution.start(args1))
+        {
+            assertTrue(run1.awaitFor(5, TimeUnit.SECONDS));
+            assertEquals(0, run1.getExitValue());
+
+            File webApp = distribution.resolveArtifact("org.eclipse.jetty.tests:test-websocket-client-webapp:war:" + jettyVersion);
+            distribution.installWarFile(webApp, "test");
+
+            int port = distribution.freePort();
+            String[] args2 = {
+                "jetty.http.port=" + port,
+                // "jetty.server.dumpAfterStart=true",
+                // "jetty.webapp.addSystemClasses+=,org.eclipse.jetty.client.",
+                // "jetty.webapp.addServerClasses+=,-org.eclipse.jetty.client.",
+                // "jetty.webapp.addSystemClasses+=,org.eclipse.jetty.util.ssl.",
+                // "jetty.webapp.addServerClasses+=,-org.eclipse.jetty.util.ssl.",
+                // "jetty.webapp.addSystemClasses+=,org.eclipse.jetty.util.component.",
+                // "jetty.webapp.addServerClasses+=,-org.eclipse.jetty.util.component."
+            };
+
+            try (DistributionTester.Run run2 = distribution.start(args2))
+            {
+                assertTrue(run2.awaitConsoleLogsFor("Started @", 10, TimeUnit.SECONDS));
+
+                // We should get the correct configuration from the jetty-websocket-httpclient.xml file.
+                startHttpClient();
+                URI serverUri = URI.create("ws://localhost:" + port + "/test");
```

Comment:
I'd like to see also a test case for `wss`, as loading the SSL classes may have a similar problem to loading the http client classes.